Feat: multiframe altda channel

samlaf opened the pull request:
<!--
Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

**Description**

Currently altda forces using "calldata" channels, which restrict their txData output to a single frame at a time. EigenDA (and probably other altdas) support larger blobs than the default 128KB frame size. This PR allows channels to have the altDA da type, which allows them to output a much larger number of frames when queried for txData. These frames are then concatenated and sent as a large blob to the altda source.

<!--
A clear and concise description of the features you're adding in this pull request.
-->

**Tests**

<!--
Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.
-->

No new tests because there isn't really any change in behaviour here since this is a small repurpose of frame concatenation logic that was already there (curious to see if the ci here will pass). The derivation pipeline also doesn't need any changes because it already can read multiple concatenated frames. Perhaps an e2e test to show that an op-node is able to derive the concatenated frames would be useful though?

**Additional context**

<!--
Add any other context about the problem you're solving.
-->

An alternative here would have been to still read single frame at a time, but increase the MaxL1FrameSize variable. This approach requires more changes, and the derivation pipeline also has hardcoded limits which would have needed to change. Also keeping small frames allows fallbacking to ethDA automatically if ever needed, potentially even inside a single channel.

**Metadata**

<!-- 
Include a link to any github issues that this may close in the following form:
- Fixes #[Link to Issue]
-->

samlaf committed:
feat(batcher): multi-frame altda channels

tynes commented:
This looks like generally a good change to me, we will get around to review more thoroughly after we complete holocene code freeze next week

samlaf commented:
@tynes any updates here?

notbdu commented:
+1 for interest in this update on Polymer's side as well.

dshiell commented:
ping, gentle push to see if it would be possible to take another look at this guy

vdrg commented:
This is important for [Batched Commitments](https://github.com/ethereum-optimism/SUPs/pull/1) as well :) @samlaf do you think it would be good to have a new `DaTypeBatchedAltDA` type? or would it be better to have an additional config option somewhere else?

samlaf committed:
ci: migrate from circleci to github actions (#20)

* ci: add test-golang workflow

* ci: delete circleci yml file

This was used by op, we are moving to using github actions for our fork

* ci: fix linter by using golangci-lint action

* ci: use golangci-lint v1.61 to match mise.toml requirement

* ci: use mise in ci to install correct versions of dependencies

* ci: remove setup-go step because go already installed by mise in previous step

* ci: build contracts that are needed for op-e2e tests

* ci: use caching for forge artifacts

* ci: remove slow op-program build

don't think we are using it... testing

* ci: make go-lint job run in parallel, and update version to try to fix errors

* ci: add go modules caching to speed up workflow

* ci: add explicit go mod download

samlaf commented:
> This is important for [Batched Commitments](https://github.com/ethereum-optimism/SUPs/pull/1) as well :) @samlaf do you think it would be good to have a new `DaTypeBatchedAltDA` type? or would it be better to have an additional config option somewhere else?

Probably yes, haven't fully thought through your proposal. Will answer in your proposal and we can continue discussion there.

samlaf committed:
feat: eigenda-holesky kurtosis devnet (#25)

* feat: add eigenda kurtosis devnet

* chore: rename altda -> eigenda devnet

* ci: add workflow to spin up eigenda kurtosis devnet

* ci: fix kurtosis-devnet.yml -- install foundry toolchain

* docs: update readme devnet name altda-devnet -> eigenda-holesky-devnet

* chore: change kt package back to ethpandaops instead of our fork (altda pr was merged)

* fix(eigenda-devnet): use teku since lighthouse is broken with minimal preset right now

* chore: bump da challenge/resolve windows 1->16 in eigenda-holesky.yaml kt-devnet file

* style: swap challenger and proposer ordering in eigenda-holesky.yaml

* ci: pin kurtosis to version 1.4.4

* ci(kurtosis-devnet): use mise to install dependencies with correct versions (kurtosis + foundry)

* ci: delete kt-install gha (we use mise install now)

* docs: explain why we use GenericCommitment and don't deploy da_challenge contract even for failover

* chore: add eigenda-holesky-devnet-clean justfile command to delete eigenda testnet enclave

samlaf committed:
docs(batcher): add documentation for DaType and txData.daType

samlaf committed:
docs: fix NextTxData comment

samlaf committed:
feat:  eigenda-memstore kurtosis devnet (#29)

* feat: add kurtosis eigenda-memstore-devnet

This devnet spins-up the eigenda-proxy in memstore mode which is much faster for testing than having to use the holesky testnet.

* chore: set da_server image to use proxy v1.6.4 instead of dev

* ci: add eigenda-memstore-devnet to ci workflow

samlaf committed:
chore(kt-devnet):  improve memstore devnet justfile and yaml (#30)

* chore(memstore-devnet): add justfile commands + fix op-geth version (bug in latest version)

* chore(kt-devnet): rename and refactor some justfile commands

* ci(kt-devnet): fix ci

1. stop running holesky devnet given holesky is borked
2. update just commands to use "-start" suffix
